### PR TITLE
msa: cleanup yaml parsing

### PR DESCRIPTION
**Pull request by rhaschke**
While tracking down a bug in yaml parsing, which finally turned out to be a bug in yamp-cpp itself (https://github.com/jbeder/yaml-cpp/pull/568), I cleaned the code a little bit.

### CODE REVIEW

File: moveit_setup_assistant/src/tools/moveit_config_data.cpp [hunk 1]
```diff
@@ -129,8 +127,7 @@ void operator>>(const YAML::Node& node, T& i)
 typedef const YAML::Node* yaml_node_t;
 
 // Helper function to find a value (yaml-cpp 0.3)
```
Comment (davetcoleman) [inline moveit_setup_assistant/src/tools/moveit_config_data.cpp]: can we just remove support for yaml-cpp 0.3 at this point?

Comment (rhaschke) [reply moveit_setup_assistant/src/tools/moveit_config_data.cpp]: yaml-cpp 0.3 is default for Ubuntu 14.04 / ROS Indigo. In the Kinetic branch, indeed we could get rid of 0.3 support.

**Comment by rhaschke** [comment]
@davetcoleman I removed yaml 0.3 support and further simplified the yaml parsing code.

File: moveit_setup_assistant/src/tools/moveit_config_data.cpp (hunk 1 above)
Comment (davetcoleman) [reply moveit_setup_assistant/src/tools/moveit_config_data.cpp]: thanks!

**Pull request merged**